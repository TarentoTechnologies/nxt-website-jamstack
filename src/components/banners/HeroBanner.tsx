import React from "react";

import {
  heroBannerStyles,
  logoHeadStyles,
  logoMainStyles,
  logoSection,
} from "./HeroBanner.module.css";

interface HeroBannerProps {
  data: any;
}

export const HeroBanner = ({ data }: HeroBannerProps) => {
  return (
    <div
      className={`container-fluid ${heroBannerStyles}`}
      data-testid="hero-banner"
    >
      <div className={`${logoSection}`}>
        <div className={`${logoMainStyles}`}>
          <img
            src={data.PrimaryImage?.localFile?.url}
            alt={data.PrimaryImage?.alternativeText}
            title={data.PrimaryImage?.caption}
            className="img-fluid"
          />
        </div>
        <div className={`${logoHeadStyles}`}>
          <img
            src={data.SecondaryImage?.localFile?.url}
            alt={data.SecondaryImage?.alternativeText}
            title={data.SecondaryImage?.caption}
            className="img-fluid"
          />
        </div>
      </div>
    </div>
  );
};
