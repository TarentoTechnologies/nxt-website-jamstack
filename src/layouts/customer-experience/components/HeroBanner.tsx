import { cardHeader } from "../../../styles/style-guide/Typography.module.css";
import {
  bannerStyles,
  border,
  headerInfo,
  imgStyles,
  subtextStyles,
  titleStyles,
} from "../CustomerExperience.module.css";

interface HeroBannerProps {
  data: {
    title: string;
    img: any;
    imgAltText?: string;
    subtext?: string;
    description: string;
  };
}

export const HeroBanner = ({ data }: HeroBannerProps) => {
  return (
    <div
      className={`container-fluid ${bannerStyles} d-flex justify-content-center`}
    >
      <div className={`row containerService paddingLeftRight15`}>
        <div
          className={`col-sm-12 col-md-6 d-flex flex-column justify-content-center px-md-0`}
        >
          <div className={`${headerInfo}`}>
            <div className={`${cardHeader} ${titleStyles}`}>{data?.title}</div>
            <div className={`${subtextStyles}`}>
              <span className={`${border}`}></span>
              <p>{data?.subtext}</p>
              <p>{data?.description}</p>
            </div>
          </div>
        </div>
        <div className={`col-sm-12 col-md-6 d-flex align-items-center`}>
          <img
            className={`${imgStyles} img-fluid`}
            src={data?.img}
            alt={data?.imgAltText}
            title={data?.imgAltText}
          />
        </div>
      </div>
    </div>
  );
};
