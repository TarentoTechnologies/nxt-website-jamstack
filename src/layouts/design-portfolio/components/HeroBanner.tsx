import { HeroBannerForTDI } from "../../../components/banners/HeroBannerForTDI";
import { PortfolioHeroBanner } from "../../../components/banners/PortfolioHeroBanner";

interface HeroBannerProps {
  heroBannerData: any;
  isImg?: boolean;
  aboutSectionData?: any;
}

export const HeroBanner = ({
  heroBannerData,
  isImg,
  aboutSectionData,
}: HeroBannerProps) => {
  return (
    <PortfolioHeroBanner
      title={heroBannerData?.Title}
      description={heroBannerData?.Desc}
      img={heroBannerData?.Image}
      isImage={isImg}
      isAgencyDetail={heroBannerData?.isAgencyDetail}
      isListingPage={heroBannerData?.isListingPage}
      withBreadcrumbs={heroBannerData?.withBreadCrumbs}
      breadCrumbLabel={heroBannerData?.BreadCrumbLabel}
      breadCrumbLink={heroBannerData?.BreadCrumbLink}
      AboutSectionData={aboutSectionData}
    />
  );
};
