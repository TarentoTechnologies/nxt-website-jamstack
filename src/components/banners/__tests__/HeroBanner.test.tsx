import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { HeroBanner } from "../HeroBanner";

const data = {
  PrimaryImage: {
    alternativeText: "NXT Logo",
    caption: "NXT Logo",
    id: "a02eb78b-c216-5263-86ca-bd30c281f873",
    localFile: {
      url: "https://strapi-stage-nxt.tarento.com/uploads/logo_main_b45ca700bb.svg",
    },
  },
  SecondaryImage: {
    alternativeText: "NXT logo description",
    caption: "NXT logo description",
    id: "b6e466bf-0c08-5cd8-b72c-edcd2b17c642",
    localFile: {
      url: "https://strapi-stage-nxt.tarento.com/uploads/logo_head_b1dd3ba367.svg",
    },
  },
};

describe("Test if the HeroBanner component is working as expected", () => {
  it("<HeroBanner /> matches snapshot", () => {
    const component = render(<HeroBanner data={data} />);
    expect(component.container).toMatchSnapshot();
  });

  it("<HeroBanner /> renders without exploding", () => {
    const component = render(<HeroBanner data={data} />);
    expect(component.getByTestId("hero-banner")).toBeInTheDocument();
  });
});
