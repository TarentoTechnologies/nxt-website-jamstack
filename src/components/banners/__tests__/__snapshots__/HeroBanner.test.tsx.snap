// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the HeroBanner component is working as expected <HeroBanner /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid heroBannerStyles"
    data-testid="hero-banner"
  >
    <div
      class="logoSection"
    >
      <div
        class="logoMainStyles"
      >
        <img
          alt="NXT Logo"
          class="img-fluid"
          src="https://strapi-stage-nxt.tarento.com/uploads/logo_main_b45ca700bb.svg"
          title="NXT Logo"
        />
      </div>
      <div
        class="logoHeadStyles"
      >
        <img
          alt="NXT logo description"
          class="img-fluid"
          src="https://strapi-stage-nxt.tarento.com/uploads/logo_head_b1dd3ba367.svg"
          title="NXT logo description"
        />
      </div>
    </div>
  </div>
</div>
`;
